### PR TITLE
Add config file support for ElasticIP, AutoScalingGroup, LaunchConfiguration, EC2

### PR DESCRIPTION
**Pull request by brandonstrohmeyer**
Adds `names_regex` config file support for:
- ElasticIP
- AutoScalingGroup
- LaunchConfiguration
- EC2 Instances

### CODE REVIEW

File: aws/eip.go [hunk 1]
```diff
@@ -70,15 +71,35 @@ func getAllEIPAddresses(session *session.Session, region string, excludeAfter ti
...
+		return false
+	}
+
+	if excludeAfter.Before(firstSeenTime) {
+		return false
+	}
+
+	// If Name is unset, GetEC2ResourceNameTagValue returns error and zero value string
+	// Ignore this error and pass empty string to config.ShouldInclude
+	allocationName, _ := GetEC2ResourceNameTagValue(address.Tags)
```
Comment (tonerdo) [inline aws/eip.go:94]: We should be using the allocation Ids and not the name

**Review by tonerdo — CHANGES_REQUESTED** [review]
Looks mostly good. I requested changes to how to decide whether or not to include EIP addresses

**Comment by brandonstrohmeyer** [comment]
Thanks for the review,

It should be noted that the EC2 Instance filter also looks at instance name and not instance ID. Additionally, the NAT Gateway filter does as well and is a past PR submission of mine (!266), it was not challenged therefore I didn't think it was an issue.

@tonerdo Is this a topic that is up for discussion? In my experience with Cloud Nuke there is _significant_ utility in being able to use a regex pattern to match a resource name and not just specify individual resource IDs. Especially in an IaC controlled environment where resources may be spun up and down, changing the underlying instance ID but retaining the same Name value. 

If this is a hard rule not up for discussion then I will make the requested changes and also open an issue to request README updates that clarify the "Names" pattern filter does not actually filter the resource names but instead filters the AWS Resource ID as it is not clear.

**Comment by tonerdo** [comment]
@brandonstrohmeyer you're right. I looked through the codebase and founds other examples. This PR is good to go

**Review by tonerdo — APPROVED** [review]

**Pull request merged**